tests/triaging_patients.spec.ts

```diff
@@ -0,0 +1,195 @@
+  await when_i_enter_the_note("Notes from nurse");
+  await when_i_click_on_the_option("Do not vaccinate");
+  await when_i_click_on_the_submit_button();
+  await when_i_click_on_the_tab("Triage complete");
+  await then_i_should_see_a_triage_row_for_the_patient("Blaine DuBuque", {
+    note: "Notes need triage",
+    action: "Do not vaccinate",
+    action_colour: "red",
+    tab: "Triage complete",
+  });
+
+  await when_i_click_on_the_tab("Needs triage");
+  await when_i_click_on_the_patient("Caridad Sipes");
+  await when_i_enter_the_note("Reached mother, should be able to proceed");
+  await when_i_click_on_the_option("Ready to vaccinate");
+  await when_i_click_on_the_submit_button();
+  await when_i_click_on_the_tab("Triage complete");
+  await then_i_should_see_a_triage_row_for_the_patient("Caridad Sipes", {
+    note: "Notes need triage",
+    action: "Vaccinate",
+    action_colour: "purple",
+    tab: "Triage complete",
+  });
+});
+
+export async function when_i_go_to_the_triage_page_for_the_first_session() {
+  await p.goto("/sessions/1");
+  await p.getByRole("link", { name: "Triage" }).click();
+}
+
+export async function when_i_go_back_to_the_triage_index_page() {
+  await p.getByRole("link", { name: "Back to triage" }).click();
+}
+
+export async function when_i_enter_the_note(note) {
+  await p.getByLabel("Triage notes").type(note);
+}
+
+export async function when_i_clear_the_note() {
+  await p.getByLabel("Triage notes").clear();
+}
+
+export async function when_i_click_on_the_submit_button() {
+  await p.getByRole("button", { name: "Save triage" }).click();
+}
+
+export async function then_i_should_see_the_triage_index_page() {
+  await expect(p.locator("h1")).toContainText("Triage");
+}
+
+export async function then_i_should_see_the_correct_breadcrumbs() {
+  await expect(p.locator(".nhsuk-breadcrumb__item:last-of-type")).toContainText(
+    "HPV session at St Andrew's Benn CofE (Voluntary Aided) Primary School",
+  );
+}
+
+export async function then_i_should_see_a_triage_row_for_the_patient(
+  name,
+  attributes = {},
+) {
+  const patient = { ...patient_expectations[name], ...attributes };
+  const id = patient.tab.toLowerCase().replace(/ /g, "-");
+  const row = p.locator(`#${id} tr`, { hasText: name });
+
+  if (patient.triage_reasons) {
+    for (let reason of patient.triage_reasons) {
+      await expect(row, `[${name}] Triage reason`).toContainText(reason);
+    }
+  }
+
+  await expect(row, `[${name}] Status text`).toContainText(patient.action);
+
+  const colourClass = "nhsuk-tag--" + patient.action_colour;
+  await expect(
+    row.locator("td:nth-child(3) div"),
+    `[${name}] Status colour`,
+  ).toHaveClass(new RegExp(colourClass));
+}
+
+export async function then_i_should_see_the_triage_page_for_the_patient(name) {
+  await expect(p.locator("h1"), `[${name}] Triage page title`).toContainText(
+    name,
+  );
+}
+
+export async function and_i_should_see_the_consent_section_for_the_patient(
```

Comment:
Functions like these feel to me like they're doing too much.

I think we should find some time, perhaps after the second model office session, to scale back our end to end tests and remove looping/conditionals/passed in arguments. I feel like we can lift these checks into unit/component tests and only rely on system/e2e tests to test core user journeys at a high level.

For example, in the method above, I don't think we should need to test for the colour of a tag in a system spec. That should be covered elsewhere.

Reply:
Ya, the viewing triage test is definitely a bit heavy. It did help catch out some things, though, so not all waste.